offset.go

```diff
@@ -0,0 +1,96 @@
+package vmap
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type (
+	// Duration is a time.Duration alias
+	Duration time.Duration
+
+	// Offset is a VAMP duration expressed as  hh:mm:ss
+	Offset struct {
+		timeDur  Duration
+		position OffsetPosition
+		percent  int
+	}
+
+	// OffsetPosition is for either passing start/end as the position of the ad
+	OffsetPosition string
+)
+
+const (
+	// OffsetPositionStart "start"
+	OffsetPositionStart OffsetPosition = "start"
+	// OffsetPositionEnd "end"
+	OffsetPositionEnd OffsetPosition = "end"
+)
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (off *Offset) MarshalText() ([]byte, error) {
+	if off.position == OffsetPositionStart || off.position == OffsetPositionEnd {
+		return []byte(off.position), nil
+	}
+
+	if off.percent != 0 {
+		return []byte(fmt.Sprintf("%d%%", off.percent)), nil
+	}
+
+	dur := off.timeDur
+	h := dur / Duration(time.Hour)
+	m := dur % Duration(time.Hour) / Duration(time.Minute)
+	s := dur % Duration(time.Minute) / Duration(time.Second)
+	ms := dur % Duration(time.Second) / Duration(time.Millisecond)
+	if ms == 0 {
+		return []byte(fmt.Sprintf("%02d:%02d:%02d", h, m, s)), nil
+	}
+	return []byte(fmt.Sprintf("%02d:%02d:%02d.%03d", h, m, s, ms)), nil
+}
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (off *Offset) UnmarshalText(data []byte) (err error) {
+	s := string(data)
+	switch OffsetPosition(s) {
+	case OffsetPositionStart, OffsetPositionEnd:
+		off.position = OffsetPosition(s)
+		return nil
+	}
+
+	if strings.Contains(s, "%") {
+		s = strings.Replace(s, "%", "", -1)
+		off.percent, err = strconv.Atoi(s)
+		return err
+	}
+
+	dur := &off.timeDur
+	s = strings.TrimSpace(s)
+	if s == "" || strings.ToLower(s) == "undefined" {
+		*dur = 0
+		return nil
+	}
+	parts := strings.SplitN(s, ":", 3)
```

Comment:
Was wondering what the benefit is from using `SplitN` vs. `Split here.

Reply:
this was just from the VAST library. i had originally intended to re-use their implementation for offset but ended up having to add a decent amount of code to it